modules/programs/superfile.nix

```
@@ -23,6 +24,27 @@ let
     types
     hm
     ;
+
+  pinnedFolderModule = types.submodule {
```

```suggestion
  pinnedFolderModule = types.submodule {
  freeformType = jsonFormat.type;
```

If we're going to use a generator, it makes sense to allow anything that fits the generator specification in case the upstream definition / supported properties changes. 

ie: They may only support two options now, but maybe that support something else in the future and we can allow a user to customize it without us needing to update the module.